lib/code_corps_web/controllers/user_skill_controller.ex

```diff
@@ -1,26 +1,43 @@
 defmodule CodeCorpsWeb.UserSkillController do
   use CodeCorpsWeb, :controller
-  use JaResource
 
-  import CodeCorps.Helpers.Query, only: [id_filter: 2]
+  alias CodeCorps.{UserSkill, User, Helpers.Query}
 
-  alias CodeCorps.UserSkill
+  action_fallback CodeCorpsWeb.FallbackController
+  plug CodeCorpsWeb.Plug.DataToAttributes
 
-  plug :load_resource, model: UserSkill, only: [:show], preload: [:user, :skill]
-  plug :load_and_authorize_changeset, model: UserSkill, only: [:create]
-  plug :load_and_authorize_resource, model: UserSkill, only: [:delete]
-  plug JaResource
+  @spec index(Conn.t, map) :: Conn.t
+  def index(%Conn{} = conn, %{} = params) do
+    with user_skills <- UserSkill |> Query.id_filter(params) |> Repo.all do
+      conn |> render("index.json-api", data: user_skills)
+    end
+  end
 
-  @spec model :: module
-  def model, do: CodeCorps.UserSkill
+  @spec show(Conn.t, map) :: Conn.t
+  def show(%Conn{} = conn, %{"id" => id}) do
+    with %UserSkill{} = user_skill <- UserSkill |> Repo.get(id) do
+      conn |> render("show.json-api", data: user_skill)
+    end
+  end
 
-  @spec filter(Plug.Conn.t, Ecto.Query.t, String.t, String.t) :: Ecto.Query.t
-  def filter(_conn, query, "id", id_list) do
-    query |> id_filter(id_list)
+  @spec create(Conn.t, map) :: Conn.t
+  def create(%Conn{} = conn, %{} = params) do
+    with %User{} = current_user <- conn |> Guardian.Plug.current_resource,
+      {:ok, :authorized} <- current_user |> Policy.authorize(:create, %UserSkill{}, params),
+      {:ok, %UserSkill{} = user_skill} <- %UserSkill{} |> UserSkill.create_changeset(params) |> Repo.insert
+    do
+      conn |> put_status(:created) |> render("show.json-api", data: user_skill)
+    end
   end
 
-  @spec handle_create(Plug.Conn.t, map) :: Ecto.Changeset.t
-  def handle_create(_conn, attributes) do
-    %UserSkill{} |> UserSkill.create_changeset(attributes)
+  @spec delete(Conn.t, map) :: Conn.t
+  def delete(%Conn{} = conn, %{"id" => id} = _params) do
+    with %UserSkill{} = user_skill <- UserSkill |> Repo.get(id),
+      %User{} = current_user <- conn |> Guardian.Plug.current_resource,
+      {:ok, :authorized} <- current_user |> Policy.authorize(:delete, user_skill),
+      {:ok, %UserSkill{} = _user_skill} <- user_skill |> Repo.delete
```

Comment:
@begedin getting a test failure [here](https://github.com/code-corps/code-corps-api/blob/develop/test/lib/code_corps_web/controllers/user_skill_controller_test.exs#L85)

Not really familiar with this `:track` code.  Lmk what you think here.

Reply:
You've discovered a bug we've been creating on some controllers with this switch

We have a plug, `SegmentPlugTracker`, which registers an action right before the conn sends a response.

This action figures out what happened (in this case, that a `UserSkill` was deleted) and calls tracking code for that action.

However, the "figuring out" also relied on ja_resource and canary, so in this case, it got broken and the controller test found it. Unfortunately it seems not all our controller tests have this `assert_received`, so we may have turned off tracking in some cases. I'll make sure to check.

We probably want to make that explicit as well, to avoid further issues, so I'll create an issue to do this. 

For know, to keep tracking working here, you'll need to do a `conn |> Conn.assign(:user_skill, user_skill) |> conn |> send_resp(:no_content, "")`